library/bignum.c

```diff
@@ -1706,7 +1706,7 @@ int mbedtls_mpi_exp_mod(mbedtls_mpi *X, const mbedtls_mpi *A,
      */
     const size_t x_index = 0;
     mbedtls_mpi_init(&W[x_index]);
-    mbedtls_mpi_copy(&W[x_index], X);
+    MBEDTLS_MPI_CHK(mbedtls_mpi_copy(&W[x_index], X));
```

Comment:
Consequence of failure: none, I think. We're copying the initial value of X, but that can't matter: it's the output variable. The comment above is confusing `X` the output variable with an implicit use of “X” as the accumulator in the loop.

I'll make a pull request to remove this line and fix the comment above once this one is merged.

Reply:
Actually, to facilitate reviewing and merging, can you please undo this change? It's unnecessary, and if you don't change this line then we don't need to handle the conflict between that and removing the line.

Reply:
OK. Removed.